infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/GeocodingService.kt

```diff
@@ -7,85 +7,96 @@ import fi.fta.geoviite.infra.common.TrackMeter
+        return getGeocodingContext(publishType, trackNumberId)?.getAddress(location)
+            ?.let { (address, intersect) -> if (intersect != WITHIN) null else address }
     }
 
-    fun getAddressPoints(locationTrack: LocationTrack, publishType: PublishType) =
-        addressPointService.getAddressPointCacheKey(publishType, locationTrack)
-            ?.let { context -> addressPointService.getAddressPointsInternal(context) }
+    fun getLocationTrackStartAndEnd(
+        publishType: PublishType,
+        locationTrackId: IntId<LocationTrack>,
+    ): AlignmentStartAndEnd? {
+        logger.serviceCall(
+            "getLocationTrackStartAndEnd",
+            "publishType" to publishType, "locationTrackId" to locationTrackId)
+        return locationTrackService.getWithAlignment(publishType, locationTrackId)?.let { (track, alignment) ->
+            val geocodingContext = getGeocodingContext(publishType, track.trackNumberId)
+            geocodingContext?.getStartAndEnd(alignment)
+        }
+    }
 
-    fun getTrackLocation(
-        alignmentId: IntId<LocationTrack>,
-        address: TrackMeter,
+    fun getReferenceLineStartAndEnd(
         publishType: PublishType,
-    ): AddressPoint? {
-        logger.serviceCall("getTrackLocation", "alignmentId" to alignmentId, "address" to address)
-        return addressPointService.getTrackGeocodingData(publishType, alignmentId)?.let { data ->
-            getTrackLocation(data.locationTrack, data.alignment, data.context, address, publishType)
+        referenceLineId: IntId<ReferenceLine>,
+    ): AlignmentStartAndEnd? {
+        logger.serviceCall("getReferenceLineStartAndEnd",
+            "publishType" to publishType, "referenceLineId" to referenceLineId)
+        return referenceLineService.getWithAlignment(publishType, referenceLineId)?.let { (line, alignment) ->
+            val geocodingContext = getGeocodingContext(publishType, line.trackNumberId)
+            geocodingContext?.getStartAndEnd(alignment)
         }
     }
 
     fun getTrackLocation(
-        locationTrack: LocationTrack,
-        alignment: LayoutAlignment,
-        context: GeocodingContext,
+        locationTrackId: IntId<LocationTrack>,
         address: TrackMeter,
         publishType: PublishType,
     ): AddressPoint? {
-        logger.serviceCall(
-            "getTrackLocation",
-            "locationTrack" to locationTrack.id, "alignment" to alignment.id, "address" to address
-        )
-        val startAddress = alignment.start?.let(context::getAddress)?.first
-        val endAddress = alignment.end?.let(context::getAddress)?.first
-        return if (startAddress == null || endAddress == null || address !in startAddress..endAddress) {
-            null
-        } else context.getProjectionLine(address)?.let { projectionLine ->
-            getProjectedAddressPoint(projectionLine, alignment)
+        logger.serviceCall("getTrackLocation", "locationTrackId" to locationTrackId, "address" to address)
```

Comment:
PublishTypen lokitus puuttuu

Reply:
Totta. Tälle touhulle voisi kyllä tehdä jonkun annotaatiosysteemin. Tuntuu tyhmältä kirjailla näitä joka paikkaan manuaalisesti.

Reply:
Saakohan tuota jotenkin automatisoitua. On joo vähän mälsää touhua